examples/wifinina/http-get/main.go

```diff
@@ -57,23 +55,28 @@ func setup() {
 		SCK:       machine.NINA_SCK,
 	})
 
-	adaptor = wifinina.New(spi,
+	device = wifinina.New(spi,
 		machine.NINA_CS,
 		machine.NINA_ACK,
 		machine.NINA_GPIO0,
 		machine.NINA_RESETN)
-	adaptor.Configure()
+	device.Configure().WithAccessPoint(ssid, pass, timeout)
```

Comment:
It might be better to use Configure() to match the other drivers in tinygo.org/x/drivers, but maybe that's just my preference.
If the reason for using fluent-style is clear, it is good to proceed as is.

Reply:
I am not a fan of fluent interfaces generally. In this case, it introduces a new API pattern that is not in use elsewhere in this driver. IMO a more "Go-like" AP probably would be:

```go
device.Configure()
device.UseAutoReconnect(true)
device.ConnectToAPI(ssid, pass, timeout)
```

Reply:
With this approach access point info leaks to `wifinina.Device` (low level SPI communication adapter), while I've tried to keep it in `wifinina.Driver` (implements `net.DeviceDriver`). But I'm not adamant on this. Can, perhaps, move Wifi stuff down to `wifinina.Device` and leave `wifinina.Driver` care about HTTP only.

On splitting.
Splitting it like this makes it unclear what happens when `device.ConnectToAP(ssid, pass, timeout)` is called, but not `device.UseAutoReconnect(true)`.
And vice versa: what is `device.UseAutoReconnect(true)` without `device.ConnectToAP(ssid, pass, timeout)`?

Tricky also (naming-wise) how to support multiple AP:
`device.ConnectToAP(ssid, pass, timeout)` -- will it connect directly? What if I call it multiple times with different params?

Another fact that makes it hard with naming is in wifinina context we have two types of "connections": wifi and http[s]. Feels better not mix them, for example by using "AccessPoint" or "Wifi" terms for wifi level.

Let me try and move Wifi handling to `wifinina.Device` and avoid fluent style. I'll make a separate PR.
API in that case will look something like this:
```
device.Configure()
device.AddAccessPoint(ssid, pass, timeout, autoconnect)
device.AddAccessPoint(ssid2, pass2, timeout2, autoconnect)
device.ConnectToAccessPoint() // optional
```
Call of `device.AddAccessPoint(ssid, pass, timeout, autoconnect)` adds access point to the list of known access points and last parameter enables auto[re]connect. 
Call of `device.ConnectToAccessPoint()` will explicitly connect device to AP (returns fast if already connected). And this is the function we can use in `wifinina.Driver`.

How does this sound?

Reply:
I would suggest look at the API to the MQTT driver for guidance here on how it should probably work.

Yes, `ConnectToAP()` should attempt to connect right away, and return an error if it fails. Reconnect logic should be to handle reconnections, not the initial connection.

If `device.autoReconnect` defaults to false, then if the connection fails then it has failed, and the various functions would return an error.

If `device.autoReconnect` is set to `true` via a call to `device.UseAutoReconnect(true)` then if the connection fails then it can try the reconnect logic.

As far as multiple AP, that would be best handled by the user's own logic. This is a case where `device.autoReconnect` being set to `false` seems like the least magic.